Provide official Docker image for murmur

It would be nice to have an official image of murmur for containerized deployments.  I can take a crack at this if I get some guidance for what version to target (1.2.x or 1.3.x)

It looks like for `static` builds, [src/murmur/murmur_ice/murmur_ice.pro](https://github.com/mumble-voip/mumble/blob/master/src/murmur/murmur_ice/murmur_ice.pro#L59) always expects slice2cpp to be at `/opt/Ice-3.3/bin/slice2cpp`, throwing an error when it isn't there.  Is there a reason for static builds hard-coding this?

```
Step 7/7 : RUN make release
 ---> Running in a159492cb0ff
cd src/mumble_proto/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/mumble_proto/mumble_proto.pro CONFIG+=no-client CONFIG+=grpc CONFIG+=static ) && make -f Makefile release
make[1]: Entering directory '/root/mumble/src/mumble_proto'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/mumble_proto'
/usr/bin/protoc --cpp_out=. -I. -I.. -I.. ../Mumble.proto
g++ -c -pipe -fvisibility=hidden -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -O2 -fPIC -std=gnu++1z -w -pthread -DRESTRICT=__restrict__ -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o ../../release/.obj/mumble_proto/Mumble.pb.o Mumble.pb.cc
rm -f ../../release/libmumble_proto.a
ar cqs ../../release/libmumble_proto.a ../../release/.obj/mumble_proto/Mumble.pb.o
make[2]: Leaving directory '/root/mumble/src/mumble_proto'
make[1]: Leaving directory '/root/mumble/src/mumble_proto'
cd src/murmur/murmur_ice/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/murmur/murmur_ice/murmur_ice.pro CONFIG+=no-client CONFIG+=grpc CONFIG+=static ) && make -f Makefile release
make[1]: Entering directory '/root/mumble/src/murmur/murmur_ice'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/murmur/murmur_ice'
/opt/Ice-3.3/bin/slice2cpp --checksum -I/opt/Ice-3.3/slice ../Murmur.ice
make[2]: /opt/Ice-3.3/bin/slice2cpp: Command not found
make[2]: *** [Makefile.Release:188: Murmur.cpp] Error 127
make[2]: Leaving directory '/root/mumble/src/murmur/murmur_ice'
make[1]: Leaving directory '/root/mumble/src/murmur/murmur_ice'
make[1]: *** [Makefile:42: release] Error 2
make: *** [Makefile:451: sub-src-murmur-murmur_ice-release_ordered] Error 2
The command '/bin/sh -c make release' returned a non-zero code: 2
```
Ubuntu installs it at `/usr/bin/slice2cpp`

I'm running into an issue building the GRPC support:
```Step 7/7 : RUN make
 ---> Running in bc9370b71c51
cd src/mumble_proto/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/mumble_proto/mumble_proto.pro 'CONFIG+=no-client grpc static' ) && make -f Makefile
make[1]: Entering directory '/root/mumble/src/mumble_proto'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/mumble_proto'
/usr/bin/protoc --cpp_out=. -I. -I.. -I.. ../Mumble.proto
g++ -c -pipe -fvisibility=hidden -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -O2 -fPIC -std=gnu++1z -w -pthread -DRESTRICT=__restrict__ -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o ../../release/.obj/mumble_proto/Mumble.pb.o Mumble.pb.cc
rm -f ../../release/libmumble_proto.a
ar cqs ../../release/libmumble_proto.a ../../release/.obj/mumble_proto/Mumble.pb.o
make[2]: Leaving directory '/root/mumble/src/mumble_proto'
make[1]: Leaving directory '/root/mumble/src/mumble_proto'
cd src/murmur/murmur_ice/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/murmur/murmur_ice/murmur_ice.pro 'CONFIG+=no-client grpc static' ) && make -f Makefile
make[1]: Entering directory '/root/mumble/src/murmur/murmur_ice'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/murmur/murmur_ice'
slice2cpp --checksum -I/usr/local/share/Ice -I/usr/share/Ice/slice -I/usr/share/ice/slice -I/usr/share/slice -I/usr/share/Ice-3.7.2/slice/ ../Murmur.ice
g++ -c -pipe -fvisibility=hidden -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -O2 -fPIC -std=gnu++1z -w -DRESTRICT=__restrict__ -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o ../../../release/.obj/murmur_ice/Murmur.o Murmur.cpp
rm -f ../../../release/libmurmur_ice.a
ar cqs ../../../release/libmurmur_ice.a ../../../release/.obj/murmur_ice/Murmur.o
make[2]: Leaving directory '/root/mumble/src/murmur/murmur_ice'
make[1]: Leaving directory '/root/mumble/src/murmur/murmur_ice'
cd src/murmur_grpcwrapper_protoc_plugin/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/murmur_grpcwrapper_protoc_plugin/murmur_grpcwrapper_protoc_plugin.pro 'CONFIG+=no-client grpc static' ) && make -f Makefile
make[1]: Entering directory '/root/mumble/src/murmur_grpcwrapper_protoc_plugin'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/murmur_grpcwrapper_protoc_plugin'
g++ -c -pipe -fvisibility=hidden -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I../mumble_proto -isystem ../mumble_proto -O2 -std=gnu++1z -pthread -Wall -W -D_REENTRANT -fPIC -DRESTRICT=__restrict__ -DMUMBLE_VERSION_STRING=1.3.0 -I. -I../../src -I. -I../mumble_proto -I../../3rdparty/arc4random-src -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o ../../release/.obj/murmur_grpcwrapper_protoc_plugin/main.o main.cpp
g++ -Wl,-z,relro -Wl,-z,now -Wl,-O1 -o ../../release/protoc-gen-murmur-grpcwrapper ../../release/.obj/murmur_grpcwrapper_protoc_plugin/main.o   -L/root/mumble/release -L/usr/lib/x86_64-linux-gnu/c++11 -lprotoc -lssl -lcrypto -ldl -lprotobuf -pthread -lz -lpthread
make[2]: Leaving directory '/root/mumble/src/murmur_grpcwrapper_protoc_plugin'
make[1]: Leaving directory '/root/mumble/src/murmur_grpcwrapper_protoc_plugin'
cd src/murmur/murmur_grpc/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /root/mumble/src/murmur/murmur_grpc/murmur_grpc.pro 'CONFIG+=no-client grpc static' ) && make -f Makefile
make[1]: Entering directory '/root/mumble/src/murmur/murmur_grpc'
make -f Makefile.Release
make[2]: Entering directory '/root/mumble/src/murmur/murmur_grpc'
/usr/bin/protoc --cpp_out=. -I. -I.. ../MurmurRPC.proto
/usr/bin/protoc --grpc_out=. --plugin=protoc-gen-grpc= -I. -I.. ../MurmurRPC.proto
: program not found or is not executable
--grpc_out: protoc-gen-grpc: Plugin failed with status code 1.
make[2]: *** [Makefile.Release:208: MurmurRPC.grpc.pb.cc] Error 1
make[2]: Leaving directory '/root/mumble/src/murmur/murmur_grpc'
make[1]: Leaving directory '/root/mumble/src/murmur/murmur_grpc'
make[1]: *** [Makefile:42: release] Error 2
make: *** [Makefile:180: sub-src-murmur-murmur_grpc-make_first-ordered] Error 2
The command '/bin/sh -c make' returned a non-zero code: 2```

It looks like it can't find the `protoc-gen-grpc` plugin.

For reference, here's the Dockerfile I'm currently using:
```
FROM ubuntu:disco

# needed to install tzdata in disco
ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get update && apt-get install -y \
    build-essential \
    pkg-config \
    qt5-default \
    libboost-dev \
    libasound2-dev \
    libssl-dev \
    libspeechd-dev \
    libzeroc-ice-dev \
    libpulse-dev \
    libcap-dev \
    libprotobuf-dev \
    protobuf-compiler \
    libprotoc-dev \
    libogg-dev \
    libavahi-compat-libdnssd-dev \
    libsndfile1-dev \
    libgrpc++-dev \
    libxi-dev \
    qtcreator

COPY . /root/mumble
WORKDIR /root/mumble

RUN qmake -recursive main.pro CONFIG+="no-client grpc static"
RUN make
```
I had to add `libprotoc-dev` to the dependency list (originally from https://wiki.mumble.info/wiki/BuildingLinux#1.3.x).

Latest issue is related to building a static binary:

```
...
usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libIce.a(DynamicLibrary.o): in function `IceInternal::DynamicLibrary::load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
(.text+0x195): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: cannot find -lcares
/usr/bin/ld: cannot find -lQt5Network
/usr/bin/ld: cannot find -lQt5Sql
/usr/bin/ld: cannot find -lQt5Xml
/usr/bin/ld: cannot find -lQt5DBus
/usr/bin/ld: cannot find -lQt5Core
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile.Release:359: ../../release/murmurd] Error 1
make[2]: Leaving directory '/root/mumble/src/murmur'
make[1]: *** [Makefile:42: release] Error 2
make[1]: Leaving directory '/root/mumble/src/murmur'
make: *** [Makefile:223: sub-src-murmur-make_first-ordered] Error 2
The command '/bin/sh -c make' returned a non-zero code: 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,63 @@
+FROM ubuntu:disco
+
+# needed to install tzdata in disco
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install -y \
+	build-essential \
+	pkg-config \
+	qt5-default \
+	libboost-dev \
+	libasound2-dev \
+	libssl-dev \
+	libspeechd-dev \
+	libzeroc-ice-dev \
+	libpulse-dev \
+	libcap-dev \
+	libprotobuf-dev \
+	protobuf-compiler \
+	protobuf-compiler-grpc \
+	libprotoc-dev \
+	libogg-dev \
+	libavahi-compat-libdnssd-dev \
+	libsndfile1-dev \
+	libgrpc++-dev \
+	libxi-dev \
+	libbz2-dev \
+	qtcreator
+
+COPY . /root/mumble
+WORKDIR /root/mumble
+
+RUN qmake -recursive main.pro CONFIG+="no-client grpc"
+RUN make release
+
+# Clean distribution stage
+FROM ubuntu:disco
+
+RUN adduser murmur
+RUN apt-get update && apt-get install -y \
+	libcap2 \
+	libzeroc-ice3.7 \
+	libprotobuf17 \
+	libgrpc6 \
+	libgrpc++1 \
+	libavahi-compat-libdnssd1 \
+	libqt5core5a \
+	libqt5network5 \
+	libqt5sql5 \
+	libqt5xml5 \
+	libqt5dbus5 \
+	&& rm -rf /var/lib/apt/lists/*
+
+COPY --from=0 /root/mumble/release/murmurd /usr/bin/murmurd
+COPY --from=0 /root/mumble/scripts/murmur.ini /etc/murmur/murmur.ini
+
+RUN mkdir /var/lib/murmur && \
+	chown murmur:murmur /var/lib/murmur && \
+	sed -i 's/^database=$/database=\/var\/lib\/murmur\/murmur.sqlite/' /etc/murmur/murmur.ini
+
+EXPOSE 64738/tcp 64738/udp 50051
+USER murmur
+
+CMD /usr/bin/murmurd -v -fg -ini /etc/murmur/murmur.ini

--- a/src/murmur/murmur.pro
+++ b/src/murmur/murmur.pro
@@ -136,8 +136,6 @@ ice {
   }
 
   unix:!macx:CONFIG(static) {
-    INCLUDEPATH *= /opt/Ice-3.3/include
-    QMAKE_LIBDIR *= /opt/Ice-3.3/lib
     LIBS *= -lbz2
     QMAKE_CXXFLAGS *= -fPIC
   }

--- a/src/murmur/murmur_ice/murmur_ice.pro
+++ b/src/murmur/murmur_ice/murmur_ice.pro
@@ -56,11 +56,6 @@ unix:!macx:CONFIG(buildenv) {
   slice.commands = $$(MUMBLE_ICE_PREFIX)/bin/slice2cpp --checksum -I$$(MUMBLE_ICE_PREFIX)/slice/ ../Murmur.ice
 }
 
-unix:!macx:CONFIG(static) {
-  INCLUDEPATH *= /opt/Ice-3.3/include
-  slice.commands = /opt/Ice-3.3/bin/slice2cpp --checksum -I/opt/Ice-3.3/slice ../Murmur.ice
-}
-
 macx:CONFIG(static) {
   LIBS *= -lbz2 -liconv
   QMAKE_CXXFLAGS *= -fPIC

